src/lib/components/Identifiers/PIDField.js

```diff
@@ -0,0 +1,506 @@
+const UPDATE_PID_DEBOUNCE_MS = 200;
+
+/**
+ * Button component to reserve a PID.
+ */
+class BtnReservePID extends Component {
+  handleClickReservePID = () => {
+    const { onPIDReserved } = this.props;
+    //PIDS-FIXME
+    console.log('TODO implement reserve PID');
+    const value = '10.1234/as7r3-234f3';
+    onPIDReserved(value);
+  };
+
+  render() {
+    const { disabled, label } = this.props;
+    return (
+      <Form.Button
+        color="green"
+        size="mini"
+        disabled={disabled}
+        onClick={this.handleClickReservePID}
+        content={label}
+      />
+    );
+  }
+}
+
+BtnReservePID.propTypes = {
+  disabled: PropTypes.bool,
+  label: PropTypes.string.isRequired,
+  onPIDReserved: PropTypes.func.isRequired,
+};
+
+BtnReservePID.defaultProps = {
+  disabled: false,
+};
+
+/**
+ * Button component to unreserve a PID.
+ */
+class BtnUnreservePID extends Component {
+  unreserve = (identifier) => {
+    const { onPIDUnreserved } = this.props;
+    //PIDS-FIXME
+    console.log('TODO implement unreserve PID', identifier);
+    onPIDUnreserved();
+  };
+
+  render() {
+    const { identifier } = this.props;
+    return (
+      <Form.Button icon="close" onClick={() => this.unreserve(identifier)} />
+    );
+  }
+}
+
+BtnUnreservePID.propTypes = {
+  identifier: PropTypes.string.isRequired,
+  onPIDUnreserved: PropTypes.func.isRequired,
+};
+
+/**
+ * Manage radio buttons choices between managed
+ * and unmanaged PID.
+ */
+class ManagedUnmanagedSwitch extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isManagedSelected: false,
+    };
+  }
+
+  handleChange = (e, { value }) => {
+    const { onManagedUnmanagedChange } = this.props;
+    const isManagedSelected = value === 'managed' ? true : false;
+    this.setState({ isManagedSelected: isManagedSelected });
+    onManagedUnmanagedChange(isManagedSelected);
+  };
+
+  render() {
+    const { isManagedSelected } = this.state;
+    const { pidLabel } = this.props;
+
+    return (
+      <Form.Group inline>
+        <Form.Field>
+          Do you already have a {pidLabel} for this document?
+        </Form.Field>
+        <Form.Field width={2}>
+          <Radio
+            label="Yes"
+            name="radioGroup"
+            value="unmanaged"
+            checked={!isManagedSelected}
+            onChange={this.handleChange}
+          />
+        </Form.Field>
+        <Form.Field width={2}>
+          <Radio
+            label="No"
+            name="radioGroup"
+            value="managed"
+            checked={isManagedSelected}
+            onChange={this.handleChange}
+          />
+        </Form.Field>
+      </Form.Group>
+    );
+  }
+}
+
+ManagedUnmanagedSwitch.propTypes = {
+  onManagedUnmanagedChange: PropTypes.func.isRequired,
+};
+
+/**
+ * Render identifier field and reserve/unreserve
+ * button components for managed PID.
+ */
+class ManagedIdentifierCmp extends Component {
```

Comment:
Are we going for brief naming e.g `Component -> Cmp` or `Button -> Btn`? I am fine with it as long as we stick to something common :)

Reply:
Went for `Cmp`, `Btn`